### PR TITLE
VC mut tab + new entry point + disable VC in data download + query multi-study from study view

### PR DESCRIPTION
**Pull request by yichaoS**
Heroku demo: https://cbioportal-pr-2151.herokuapp.com/

# What? Why?
Fix https://github.com/cBioPortal/iViz/issues/373
Fix https://github.com/cBioPortal/iViz/issues/356
Fix https://github.com/cBioPortal/iViz/issues/376
Fix https://github.com/cBioPortal/iViz/issues/358
Fix https://github.com/cBioPortal/iViz/issues/383
Fix https://github.com/cBioPortal/iViz/issues/384

screen shot 2017-02-20 at 3 31 32 pm

### CODE REVIEW

File: portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp [hunk 1]
```diff
@@ -128,7 +128,16 @@
                         annotation: annotationCol.render
                     },
                     columnVisibility: {
-                        annotation: 'visible'
+                        annotation: 'visible',
```
Comment (onursumer) [inline portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp]: duplicate line

File: portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp [hunk 2]
```diff
@@ -128,7 +128,16 @@
...
+                        "tumorType": function() {
```
Comment (onursumer) [inline portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp]: this function overrides the library default. if we want to keep the default behavior intact when `isVirtualStudy === false`, we still need to do something like this:

```JavaScript
"tumorType": function (util, gene) {
    if (isVirtualStudy) {
        return "visible";
    }
    else if (util.distinctTumorTypeCount(gene) > 1) {
        return "visible";
    }
    else {
        return "hidden";
    }
}
```

File: portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp (hunk 1 above)
Comment (yichaoS) [inline portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp]: ? No there was an additional comma.

File: portal/src/main/webapp/WEB-INF/jsp/cross_cancer_results.jsp [hunk 3]
```diff
@@ -179,6 +179,9 @@ if (sessionError != null) {  %>
             } else {
                 $("#modify_query_btn").addClass("active");
             }
+            if (window.studies.length > 0) {
```
Comment (zhx828) [inline portal/src/main/webapp/WEB-INF/jsp/cross_cancer_results.jsp]: Should we consider whether window has studies property?

File: portal/src/main/webapp/WEB-INF/jsp/index.jsp [hunk 4]
```diff
@@ -169,7 +169,13 @@ if (sessionError != null) {  %>
                 }
             %>
             <%@ include file="query_form.jsp" %>
-
+            <script type="text/javascript">
+                $(document).ready(function() {
+                    if (window.cancer_study_id_selected.length > 0 && window.cancer_study_id_selected !== "null") {
```
Comment (zhx828) [inline portal/src/main/webapp/WEB-INF/jsp/index.jsp]: Same applies to here, cancer_study_id_selected.length may give exception

File: portal/src/main/webapp/WEB-INF/jsp/query_form.jsp [hunk 5]
```diff
@@ -178,13 +184,18 @@
...
-        <% conditionallyOutputGenomespaceOption(localTabIndex, out); %>
+            <input type="button" id="dashboard_button" class="btn btn-default btn-lg" name="Overview" value="Overview" onclick='clickDashboard()' disabled/>
+            <button id="main_submit" class="btn btn-default btn-lg" onclick="submitHandler()" name="<%= QueryBuilder.ACTION_NAME%>" value="<%= QueryBuilder.ACTION_SUBMIT %>" title='Submit Query' disabled readonly>Query</button>
+            <% conditionallyOutputGenomespaceOption(localTabIndex, out); %>
         </p>
         </form>
     </div>
 </div>      
 
+<script>
```
Comment (kalletlak) [inline portal/src/main/webapp/WEB-INF/jsp/query_form.jsp]: Empty `<script>` block

Comment (zhx828) [inline portal/src/main/webapp/WEB-INF/jsp/query_form.jsp]: No longer needed?

File: portal/src/main/webapp/WEB-INF/jsp/index.jsp (hunk 4 above)
Comment (yichaoS) [inline portal/src/main/webapp/WEB-INF/jsp/index.jsp]: In either cases, there's no way the exception could ever happen - but yea, it doesn't make sense. (The code here is really ancient..)

File: portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp (hunk 1 above)
Comment (onursumer) [inline portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp]: There is another line (line 140) with same property: `'annotation': 'visible'`

File: portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp (hunk 2 above)
Comment (onursumer) [inline portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp]: @yichaoS are we sure that we don't want to keep the default behavior intact when `isVirtualStudy === false`?

File: portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp (hunk 1 above)
Comment (yichaoS) [inline portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp]: I see. Done. Thanks

File: portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp (hunk 2 above)
Comment (yichaoS) [inline portal/src/main/webapp/WEB-INF/jsp/mutation_details.jsp]: make sense. Done.

**Comment by kalletlak** [comment]
Opening n number of tabs when Overview button is clicked of n selected studies

**Comment by zhx828** [comment]
No sure whether this should be fixed in this pull..  but if you go back to previous page from result page, the pre-selected study is gone but the query button is enabled.
feb-27-2017 17-41-32

**Comment by kalletlak** [comment]
Better not to over populate this pr with many changes

**Comment by kalletlak** [comment]
https://github.com/cBioPortal/iViz/issues/358 isn't working for saved virtual cohorts
Example
screen shot 2017-02-27 at 6 53 57 pm

**Comment by yichaoS** [comment]
@zhx828 As long as the query form is filled with legit gene(s), the query button should be enabled. On current public portal, if the user doesn't select any studies, the query button will query the gene(s) with all studies.

**Pull request merged**